config/webpack.common.js

```diff
@@ -128,9 +128,9 @@ module.exports = {
       {
         test: /\.ts$/,
         loaders: [
+          '@angularclass/hmr-loader',
           'awesome-typescript-loader',
-          'angular2-template-loader',
-          '@angularclass/hmr-loader'
+          'angular2-template-loader'
```

Comment:
It was very confusing for me when the TypeScript compiler started to report errors in line/column position which did not even exist in the source file. It took some time for me to realize the angular2-hmr-loader injected new lines into the code.

The injected code in angular2-hmr-loader is not typed and does not use any TypeScript extension, it is plain JavaScript. Because of that, it would make more sense to apply it to the JavaScript code (output of awesome-typescript-loader) and not the TypeScript source. Why to push is through the type checker when we know it is not typed?

If I keep the original order, the angular2-hmr-loader is applied before the TypeScript compilation. In case of noImplicitAny you will have type errors in the injected code, but you do not have any control of them.

Another option is to annotate the injected code in angular2-hmr-loader with type information, but in this case it would not be anymore JavaScript compatible. However if you think that is not a problem and the target of angular2-hmr-loader is always TypeScript source, then I can make a pull request in that project to add the missing types (probably few basic any is enough though).